### DISCUSSION TITLE
Spinbox button event

### DISCUSSION THREAD

**Discussion by mtibi75** [arendst/Tasmota#20829] [Support request] [answered]
Does anyone have a working sample program?
(https://docs.lvgl.io/8.3/widgets/extra/spinbox.html#example)
(https://tasmota.github.io/docs/LVGL_API_Reference/#class-lvevent)
(13.3.0(TasmoCompiler-esp32s3))

def increment_event_cb(e)
code = e.get_code()
 if code == lv.EVENT_SHORT_CLICKED | code == lv.EVENT_LONG_PRESSED_REPEAT
  spinbox.increment()
 end
end
inc_btn.add_event_cb(increment_event_cb, lv.EVENT_ALL, 0)

error message is:
BRY: Exception> 'attribute_error' - class 'lv_btn' has no method 'get_code'

THX

**Comment by sfromis** [comment]
Better upgrade to current Tasmota version, instead of the older 13.3, recent updates have happened around LVGL support.

**Reply by mtibi75** [reply]
I upgraded to 13.4.0.1(TasmoCompiler-esp32s3).
Unfortunately it doesn't work: BRY: Exception> 'attribute_error' - the 'lv_button' object has no attribute 'get_code'
and there were many other error messages.

**Reply by s-hadinger** [reply]
Since you don't share your code nor the logs, we cannot guess what's wrong.

**Reply by s-hadinger** [reply]
As mentioned in the example, the event_cb receives the object as first argument, then the event itself.

The following should work (although not tested)
```
def increment_event_cb(obj, e)
  code = e.get_code()
  if code == lv.EVENT_SHORT_CLICKED || code == lv.EVENT_LONG_PRESSED_REPEAT
    spinbox.increment()
  end
end
inc_btn.add_event_cb(increment_event_cb, lv.EVENT_ALL, 0)
```

**Reply by sfromis** [reply, marked as answer]
To be valid without depending on a global name, I think that it should be `var code = e.get_code()`

**Reply by mtibi75** [reply]
Both of you are right.
This is working, but i'm not tested with ts, no error message:

def increment_event_cb(obj, e)
  var code = e.get_code()
  if code == lv.EVENT_SHORT_CLICKED || code == lv.EVENT_LONG_PRESSED_REPEAT
    spinbox.increment()
  end
end

Thank you!
